metricbeat/module/vsphere/resourcepool/resourcepool.go

```diff
@@ -0,0 +1,230 @@
+
+	v, err := mgr.CreateContainerView(ctx, c.ServiceContent.RootFolder, []string{"ResourcePool"}, true)
+	if err != nil {
+		return fmt.Errorf("error in CreateContainerView: %w", err)
+	}
+
+	defer func() {
+		if err := v.Destroy(ctx); err != nil {
+			m.Logger().Errorf("error trying to destroy view from vSphere: %w", err)
+		}
+	}()
+
+	// Retrieve property for all ResourcePools.
+	var rps []mo.ResourcePool
+	err = v.Retrieve(ctx, []string{"ResourcePool"}, []string{"name", "summary", "overallStatus", "vm"}, &rps)
+	if err != nil {
+		return fmt.Errorf("error in Retrieve: %w", err)
+	}
+
+	// Create a performance manager
+	perfManager := performance.NewManager(c)
+
+	// Retrieve metric IDs for the specified metric names
+	metrics, err := perfManager.CounterInfoByName(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve metrics: %w", err)
+	}
+
+	// Retrieve only the required metrics
+	requiredMetrics := make(map[string]*types.PerfCounterInfo)
+
+	var spec types.PerfQuerySpec
+
+	for _, name := range metricNames {
+		metric, exists := metrics[name]
+		if !exists {
+			m.Logger().Warnf("Metric %s not found", name)
+			continue
+		}
+		requiredMetrics[name] = metric
+	}
+
+	metricIDs := make([]types.PerfMetricId, 0, len(requiredMetrics))
+	for _, metric := range requiredMetrics {
+		metricIDs = append(metricIDs, types.PerfMetricId{
+			CounterId: metric.Key,
+		})
+	}
+	pc := property.DefaultCollector(c)
+	for i := range rps {
+
+		metricMap := map[string]interface{}{}
+
+		assetNames, err := getAssetNames(ctx, pc, &rps[i])
+		if err != nil {
+			m.Logger().Errorf("Failed to retrieve object from resource pool %s: %w", rps[i].Name, err)
+		}
+
+		spec = types.PerfQuerySpec{
+			Entity:     rps[i].Reference(),
+			MetricId:   metricIDs,
+			MaxSample:  1,
+			IntervalId: 20, // right now we are only grabbing real time metrics from the performance manager
+		}
+
+		// Query performance data
+		samples, err := perfManager.Query(ctx, []types.PerfQuerySpec{spec})
+		if err != nil {
+			m.Logger().Errorf("Failed to query performance data from resource pool %s: %v", rps[i].Name, err)
+			continue
+		}
+
+		if len(samples) == 0 {
+			m.Logger().Debug("No samples returned from performance manager")
+			continue
+		}
+
+		results, err := perfManager.ToMetricSeries(ctx, samples)
+		if err != nil {
+			m.Logger().Errorf("Failed to convert performance data to metric series for resource pool %s: %v", rps[i].Name, err)
+		}
+
+		for _, result := range results[0].Value {
+			if len(result.Value) > 0 {
+				metricMap[result.Name] = result.Value[0]
+				continue
+			}
+			m.Logger().Debugf("For resource pool %s,Metric %v: No result found", rps[i].Name, result.Name)
+		}
+
+		reporter.Event(mb.Event{
+			MetricSetFields: m.eventMapping(rps[i], &metricData{
+				perfMetrics: metricMap,
+				assetsName:  assetNames,
+			}),
+		})
+	}
+
+	return nil
+}
+
+func getAssetNames(ctx context.Context, pc *property.Collector, rp *mo.ResourcePool) (assetNames, error) {
```

Comment:
Resourcepools are associated with Clusters as well. 
Can we retrieve that information from the cluster subobject under resourcepool ?

Reply:
we don't have subObject in resourcePool.